crates/driver/src/infra/liquidity/fetcher.rs

```diff
@@ -24,8 +25,32 @@ impl Fetcher {
     }
 
     /// Fetches all relevant liquidity for the orders.
-    pub async fn fetch(&self, orders: &[order::Order]) -> Result<Vec<liquidity::Liquidity>, Error> {
-        let liquidity = self.inner.fetch(orders).await?;
+    pub async fn for_auction(
```

Comment:
This is not a dealbreaker, but I'd design this slightly differently.

One thing that you're doing right is you're defining the infrastructural component `Fetcher` to work with domain objects, which is good. But in this case I think that the methods `for_auction` and `for_quote` are too specific to warrant being defined in the `Fetcher` type. `for_quote` belongs in the `quote` domain module, whereas `for_auction` probably belongs to `competition`.

For example, `Simulator` works with `eth::Tx` because it's a component which allows transactions to be simulated - it doesn't (and shouldn't) know what a settlement is.

The `Mempool` type, OTOH, is defined to work with a `settlement::Simulated` which is also a mistake from a puristic point of view, but it's a necessary evil because of the boundary integration :confused: But ideally `Mempool` should also accept an `eth::Tx` and not know about settlements.

Reply:
982c0efe6a358148cb29cf396b4957090df0a74e

`infra::liquidity::Fetcher` now operates on `domain::liquidity::TokenPair`, and each individual domain module is responsible for determining which token pairs matter to it for liquidity, which is much nicer IMO (so `domain::competition` computes the set of token pairs it cares about for solving, and `domain::quote` computes the set of token pairs it cares about for quoting).